buffer/src/main/java/io/netty/buffer/UnpooledUnsafeDirectByteBuf.java

```diff
@@ -364,142 +272,20 @@ public ByteBuf getBytes(int index, OutputStream out, int length) throws IOExcept
         return this;
     }
 
-    @Override
-    public int getBytes(int index, GatheringByteChannel out, int length) throws IOException {
-        return getBytes(index, out, length, false);
-    }
-
-    private int getBytes(int index, GatheringByteChannel out, int length, boolean internal) throws IOException {
-        ensureAccessible();
-        if (length == 0) {
-            return 0;
-        }
-
-        ByteBuffer tmpBuf;
-        if (internal) {
-            tmpBuf = internalNioBuffer();
-        } else {
-            tmpBuf = buffer.duplicate();
-        }
-        tmpBuf.clear().position(index).limit(index + length);
-        return out.write(tmpBuf);
-    }
-
-    @Override
-    public int getBytes(int index, FileChannel out, long position, int length) throws IOException {
-        return getBytes(index, out, position, length, false);
-    }
-
-    private int getBytes(int index, FileChannel out, long position, int length, boolean internal) throws IOException {
-        ensureAccessible();
-        if (length == 0) {
-            return 0;
-        }
-
-        ByteBuffer tmpBuf = internal ? internalNioBuffer() : buffer.duplicate();
-        tmpBuf.clear().position(index).limit(index + length);
-        return out.write(tmpBuf, position);
-    }
-
-    @Override
-    public int readBytes(GatheringByteChannel out, int length) throws IOException {
-        checkReadableBytes(length);
-        int readBytes = getBytes(readerIndex, out, length, true);
-        readerIndex += readBytes;
-        return readBytes;
-    }
-
-    @Override
-    public int readBytes(FileChannel out, long position, int length) throws IOException {
-        checkReadableBytes(length);
-        int readBytes = getBytes(readerIndex, out, position, length, true);
-        readerIndex += readBytes;
-        return readBytes;
+    void getBytes(int index, OutputStream out, int length, boolean internal) throws IOException {
```

Comment:
Should this have been an override, or should the `internal` parameter  be used?

Reply:
@carl-mastrangelo I had to study it and remind myself, but it's intentional. The `internal` param doesn't apply in the `Unsafe` case since it doesn't need to use an intermediate `ByteBuffer`. The corresponding method is directly above but I guess it still wouldn't harm to add a comment, I'll do that!

Reply:
@carl-mastrangelo actually after looking more you're right that was a small mistake (thanks!) - this method was intended to be the only overridden one, not the public one above it. Have pushed a commit to rectify.